prometheus/exporter/gpu_exporter.py

```diff
@@ -54,9 +54,12 @@ def collect_gpu_info():
     try:
         logger.info("call nvidia-smi to get gpu metrics")
 
-        smi_output = utils.check_output(["nvidia-smi", "-q", "-x"])
-
-        return parse_smi_xml_result(smi_output)
+        nvidia_smi_exists = len(utils.check_output(["which", "nvidia-smi"])) > 0
+        if nvidia_smi_exists:
+            smi_output = utils.check_output(["nvidia-smi", "-q", "-x"])
+            return parse_smi_xml_result(smi_output)
+        else:
+            logger.info('nvidia-smi not found')
```

Comment:
maybe using logger.warning.

For events we expected it happen and not happen a lot, we use info.
For events we do not expect but reasonable, we use warning.
For events indication configuration/hardware/environment error that may cause the program not working/act wrongly, we use error.

Reply:
refactored